lib/src/test/java/com/basistheory/android/view/mask/MaskTests.kt

```diff
@@ -0,0 +1,95 @@
+package com.basistheory.android.view.mask
+
+
+import org.junit.Test
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+
+class MaskTests {
+    @Test
+    fun `masks are applied correctly`() {
+        val digitRegex = Regex("""\d""")
+        val maskPattern = listOf(
+            "+",
+            "1",
+            "(",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            ")",
+            " ",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            "-",
+            digitRegex,
+            digitRegex,
+            digitRegex,
+            digitRegex
+        )
+        val mask = Mask(maskPattern)
+
+        val maskResult = mask.apply(
+            "2345678900",
+            Action.INSERT
+        )
+
+        expectThat(
+            maskResult.maskedValue
+        ).isEqualTo("+1(234) 567-8900")
+        expectThat(
+            maskResult.unMaskedValue
+        ).isEqualTo("2345678900")
+        expectThat(
+            maskResult.isDone
+        ).isEqualTo(true)
```

Comment:
There's some fancy syntax supported by this assertion library (Strikt) to combine all these assertions together - it gives you better feedback if multiple assertions fail:

```kotlin
expectThat(maskResult.maskedValue) {
    get { maskedValue }.isEqualTo("+1(234) 567-8900")
    get { unMaskedValue }.isEqualTo("2345678900")
    get { isDone }.isTrue()
}
```

Reply:
Updated accordingly